include/open62541/plugin/securitypolicy.h

```diff
@@ -305,6 +305,9 @@ struct UA_SecurityPolicy {
      * depends on the used key length. */
     UA_ByteString localCertificate;
 
+    UA_NodeId certificateGroupId;
+    UA_NodeId certificateTypeId;
```

Comment:
The member field certificateGroupId is not required here. Instead, it should be placed as parameter in any function that requires it (e.g. into createSigningRequest). This avoids having to duplicate Security Policies for each Certificate Group where it shall be used and it also avoids having to filter for Security Policies (for the DefaultApplicationGroup) when creating Endpoints. The other reasons, why I think this field is of no use, are the following:

- Security Policies are required for establishing Secure Channels and thus in the Endpoints. This use case implictly requires to use the DefaultApplicationGroup -> no need of the field here.
- Security Policies are required for validation of certificate-based user tokens. This use case implicitly requires to use the DefaultUserTokenGroup -> no need of the field here.
- Security Policies are required to validate certificates to be placed into the trust lists. The API for this always provides the target certificateGroupId which can be forwarded.
- Security Policies are required to create self-signed certificates or certificate signing request. The API for this also provides the target certificateGroupId which can be forwared.

Do I miss something?

BTW, a similiar thing is the localCertificate field, which might be the reason why the certificateGroupId field found its way here. The local certificate is not required, because it can always be read (at least now) from a CertificateGroup. And in the Endpoint case it is already a member of UA_EndpointDescription...